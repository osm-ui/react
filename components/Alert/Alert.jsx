import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styled from 'styled-components';


const contexts = [
    'info',
    'success',
    'warning',
    'danger',
];

const colorsStyle = props => contexts.reduce((reducedStyles, context) => {
    const colors = props.theme.alert[context];

    return (`
        ${reducedStyles}

        &.alert-${context} {
            color: ${colors.color};
            background-color: ${colors.backgroundColor};
            border-color: ${colors.borderColor};
        }
    `);
}, '');


const StyledDiv = styled.div`
    border-radius: ${props => props.theme.borderRadius};
    ${props => colorsStyle(props)}
`;


const Alert = ({
    context,
    ...props
}) => {
    const classes = classnames({
        alert: true,
        [`alert-${context}`]: true,
    });

    return (
        <StyledDiv className={classes} {...props} />
    );
};


Alert.propTypes = {
    context: PropTypes.oneOf(contexts),
};


Alert.defaultProps = {
    context: 'info',
};

Alert.displayName = 'Alert';

export default Alert;
