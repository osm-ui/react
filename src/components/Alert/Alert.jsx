import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styled from 'styled-components';

import { contexts } from '../../constants';

const colorsStyle = props =>
  contexts.reduce((reducedStyles, context) => {
    const colors = props.theme.alert[context];

    return `
      ${reducedStyles}

      &.alert-${context} {
          color: ${colors.color};
          background-color: ${colors.backgroundColor};
          border-color: ${colors.borderColor};

          a {
              color: ${colors.color};
              text-decoration: underline;
          }

          a:hover, a:focus, a:active {
              color: ${colors.color};
              text-decoration: none;
          }
      }
    `;
  }, '');

const StyledDiv = styled.div`
  border-radius: ${props => props.theme.borderRadius};
  ${props => colorsStyle(props)};
`;

const Alert = ({ context, className, ...props }) => {
  const classes = classnames(className, {
    alert: true,
    [`alert-${context}`]: true
  });

  return <StyledDiv className={classes} {...props} />;
};

Alert.propTypes = {
  context: PropTypes.oneOf(contexts),
  className: PropTypes.string
};

Alert.defaultProps = {
  context: 'info',
  className: ''
};

Alert.displayName = 'Alert';

export default Alert;
