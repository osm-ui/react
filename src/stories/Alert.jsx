import React from 'react';
import { storiesOf } from '@storybook/react';
import { host } from 'storybook-host';
import { withKnobs } from '@storybook/addon-knobs';

import defaultHostOptions from './defaultHostOptions';

import { DefaultTheme, Alert } from '../index';

storiesOf('Alert')
  .addDecorator(withKnobs)
  .addDecorator(
    host({
      ...defaultHostOptions,
      title: 'Alert'
    })
  )
  .addWithInfo('Context', () => (
    <DefaultTheme>
      <Alert context="success">
        Hi! This is a <u>success</u> alert!
      </Alert>
      <Alert context="info">
        Hi! This is a <u>info</u> alert!
      </Alert>
      <Alert context="warning">
        Hi! This is a <u>warning</u> alert!
      </Alert>
      <Alert context="danger">
        Hi! This is a <u>danger</u> alert!
      </Alert>
    </DefaultTheme>
  ));
