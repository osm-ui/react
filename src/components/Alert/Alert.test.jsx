import React from 'react';
import {
  snapshotWithElementChildren,
  snapshotWithTextChildren
} from '../../helpers/tests';
import Alert from './';

describe('When using snapshots', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(Alert));

  it('Should render with a text children', () =>
    snapshotWithTextChildren(Alert));
});
