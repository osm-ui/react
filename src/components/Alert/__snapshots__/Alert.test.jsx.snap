// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using snapshots Should render with a text children 1`] = `
<span
  className="sc-bdVaJa lbJuWU"
>
  <div
    className="alert alert-info sc-bwzfXH eSIoy"
  >
    This is a text
  </div>
</span>
`;

exports[`When using snapshots Should render with an element children 1`] = `
<span
  className="sc-bdVaJa lbJuWU"
>
  <div
    className="alert alert-info sc-bwzfXH eSIoy"
  >
    <div />
  </div>
</span>
`;
